plugins/plugin-chart-echarts/src/Timeseries/transformProps.ts

```diff
@@ -126,6 +125,21 @@ export default function transformProps(chartProps: ChartProps): EchartsProps {
     if (min === undefined) min = 0;
     if (max === undefined) max = 1;
   }
+
+  const xAxisFormatter = (value: number) => {
+    let dateFormatter;
+
+    if (xAxisTimeFormat === smartDateFormatter.id) {
+      dateFormatter = getTimeFormatterForGranularity(timeGrainSqla);
+    } else if (xAxisTimeFormat) {
+      dateFormatter = getTimeFormatter(xAxisTimeFormat);
+    } else {
+      dateFormatter = String;
+    }
+
+    return dateFormatter(value);
+  };
```

Comment:
It feels excessive to evaluate this every time we want to format a value. Could we not do this just once and then 
```suggestion
  let xAxisFormatter;
  if (xAxisTimeFormat === smartDateFormatter.id) {
    xAxisFormatter = getTimeFormatterForGranularity(timeGrainSqla);
  } else if (xAxisTimeFormat) {
    xAxisFormatter = getTimeFormatter(xAxisTimeFormat);
  } else {
    xAxisFormatter = String;
  }
```